src/licensedcode/required_phrases.py

```diff
@@ -884,19 +1020,30 @@ class RequiredPhraseRuleCandidate:
     raw_text = attr.ib(metadata=dict(help='Raw, original required phrase text.'))
     normalized_text = attr.ib(metadata=dict(help='Normalized required phrase text.'))
 
-    def is_good(self, rule):
+    def is_good(self, rule, min_tokens, min_single_token_len):
         """
-        Return True if this phrase is a minimally suitable to use as a required phrase
+        Return True if this phrase is a minimally suitable to use as a required phrase.
+        Use the original rule to ensure we skip when referenced_filenames could be damaged.
+        Also skip short rules that would contain stopwords as they could not be detected correctly.
```

Comment:
We should probably also check here for required phrase texts which are quite large, as these can be better handled either by minimum coverage or by skipping required phrase generation from the source rules.